src/View/StringTemplate.php

```diff
@@ -339,13 +340,16 @@ protected function _formatAttribute(string $key, $value, $escape = true): string
     /**
      * Adds a class and returns a unique list either in array or space separated
      *
-     * @param array|string $input The array or string to add the class to
-     * @param array|string $newClass the new class or classes to add
+     * @param \stdClass|array|string|bool|null $input The array or string to add the class to
+     * @param array|string|bool|null $newClass the new class or classes to add
      * @param string $useIndex if you are inputting an array with an element other than default of 'class'.
-     * @return array<string>|string
+     * @return array<string>|string|null
      */
-    public function addClass($input, $newClass, string $useIndex = 'class')
-    {
+    public function addClass(
+        stdClass|array|string|bool|null $input,
```

Comment:
LOL.. How does this get `stdClass`?

Reply:
An explicit test: https://github.com/cakephp/cakephp/blob/5.0/tests/TestCase/View/StringTemplateTest.php#L332-L333

Reply:
That's to test passing an object, not stdClass specifically. So I think we should just use `mixed` here.